gpii/node_modules/gpii-oauth2/gpii-oauth2-utilities/src/OAuth2Utilities.js

```diff
@@ -66,24 +66,29 @@ gpii.oauth2.getCurrentTimestamp = function () {
 
 /**
  * Calculate the timestamp of currentTime + expiresIn.
+ * @param timestampStarts {Date} A start timestamp in the format returned by Date()
  * @param expiresIn {Number} The number of seconds that the expiration will occur.
  * @return {String} A date in simpilified ISO string format.
  */
-gpii.oauth2.getTimestampExpires = function (expiresIn) {
-    return new Date(new Date().getTime() + expiresIn * 1000).toISOString();
+gpii.oauth2.getTimestampExpires = function (timestampStarts, expiresIn) {
+    if (!timestampStarts) {
+        return undefined;
+    }
+    return new Date(timestampStarts.getTime() + expiresIn * 1000).toISOString();
 };
 
 /**
  * Compare the current time with the expiresIn time and return the number of seconds that the expiration will occur.
+ * @param timestampStarts {Date} A start timestamp in the format returned by Date()
  * @param timestampExpires {String} A string in the format returned by Date().toISOString()
  * @return {Number} The number of seconds that the expiration will occur. Return 0 if the given timestampExpires < the current timestamp.
  */
-gpii.oauth2.getExpiresIn = function (timestampExpires) {
-    if (!timestampExpires) {
+gpii.oauth2.getExpiresIn = function (timestampStarts, timestampExpires) {
```

Comment:
I really like the new date function designs.

However, and I'm sorry that I didn't think of this earlier, I believe that `gpii.oauth2.getExpiresIn` is actually no longer needed. When we changed to always issuing a new access token rather than reusing, I think we no longer need to check the amount of time left before expiration.

Does that seem right?

Reply:
You are right that `gpii.oauth2.getExpiresIn` is no longer needed in this pull request, but in [the GPII-2538 branch](https://github.com/cindyli/universal/tree/GPII-2538) I'm currently working on where the access token needs to be validated, this function is required. I figured instead of removing it from here and resolving the conflicts when merging with GPII-2538 branch, it's easier to just keep it here.

If you feel it's more proper to add this function in the GPII-2538 branch, I'm totally fine with it too and will make the change. Let me know. Thanks.

Reply:
Cindy, if it's not too much work, would you be able to move it over to GPII-2538?

Reply:
@simonbates, this function has been moved.